libsolidity/interface/CompilerStack.cpp

```diff
@@ -696,18 +696,31 @@ bool CompilerStack::compile(State _stopAfter)
 					}
 				}
 	m_stackState = CompilationSuccessful;
-	this->link();
+	if (performLinking == true)
+		this->link();
 	return true;
 }
 
-void CompilerStack::link()
+set<string> CompilerStack::link()
 {
 	solAssert(m_stackState >= CompilationSuccessful, "");
+
+	std::set<std::string> libNames;
+
 	for (auto& contract: m_contracts)
 	{
-		contract.second.object.link(m_libraries);
+		for (auto& library: m_libraries)
+			for (auto& con: m_contracts)
+				if (library.first == con.first.substr(con.first.find_last_of('/') + 1))
+				{
+					contract.second.object.linkReferences[contract.second.object.bytecode.size()] = library.first;
```

Comment:
Even just with this line - why would you modify every contract's link references just because of the fact that we compiled it together with an unrelated library?

Reply:
Oh yeah, you are right it's useless. I will remove that foreach loop.